src/west/app/main.py

```diff
@@ -644,10 +645,34 @@ def add_argument(self, *args, **kwargs):
         # Let argparse handle the actual argument.
         super().add_argument(*args, **kwargs)
 
+    def parse_known_args(self, args=None, namespace=None):
+        # argparse will terminate the application without raising an
```

Comment:
oh dear...I think I was confused, because I didn't know that the `SystemExit` thrown by argparse doesn't inherit from `Exception`, so in my testing I concluded that argparse had a different way of terminating the program rather than an exception (which is wrong). Sorry, I'll change the way I implemented this PR to just use a standard try/except to catch (and re-raise) the `SystemExit` that occurs when `parse_known_args()` fails.